src/lib/logical_query_plan/functional_dependency.cpp

```diff
@@ -51,3 +53,13 @@ std::ostream& operator<<(std::ostream& stream, const FunctionalDependency& expre
 }
 
 }  // namespace opossum
+
+namespace std {
+
+size_t hash<opossum::FunctionalDependency>::operator()(const opossum::FunctionalDependency& fd) const {
+  auto hash = boost::hash_range(fd.determinants.cbegin(), fd.determinants.cend());
+  boost::hash_combine(hash, boost::hash_range(fd.dependents.cbegin(), fd.dependents.cend()));
```

Comment:
If one set of determinants determines multiple dependents, they should be grouped in a single FD. It's not an issue if they are not, but I'd expect the common case to be that they are. As such, including the dependents in the hash might be expensive without adding too much to the collision safety.

Reply:
Both has been implemented now.